src/FakeItEasy/Core/ArgumentConstraintTrap.cs

```diff
@@ -7,7 +7,7 @@ internal class ArgumentConstraintTrap
         : IArgumentConstraintTrapper
     {
         [ThreadStatic]
-        private static List<IArgumentConstraint> trappedConstraints;
+        private static SynchronizedCollection<IArgumentConstraint> trappedConstraints;

```

Comment:
This collection will only ever be used from a single thread, since it's `ThreadStatic`